packages/php/src/LingoDotDevEngine.php

```diff
@@ -0,0 +1,484 @@
+<?php
+
+namespace LingoDotDev\PhpSdk;
+
+use Visus\Cuid2\Cuid2;
+use GuzzleHttp\Client as HttpClient;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Utils;
+// Removed: use Symfony\Component\DomCrawler\Crawler;
+use InvalidArgumentException;
+use RuntimeException;
+use Throwable; // Import Throwable for broader exception catching in batch
+
+class LingoDotDevEngine
+{
+    protected array $config;
+    protected HttpClient $httpClient;
+
+    /**
+     * Default configuration values.
+     */
+    private const DEFAULTS = [
+        "apiUrl" => "https://engine.lingo.dev",
+        "batchSize" => 25,
+        "idealBatchItemSize" => 250,
+    ];
+
+    // Removed LOCALIZABLE_ATTRIBUTES and UNLOCALIZABLE_TAGS constants
+
+    /**
+     * Create a new LingoDotDevEngine instance.
+     *
+     * @param array $config Configuration options for the Engine.
+     *                      Required: "apiKey" (string)
+     *                      Optional: "apiUrl" (string, default: "https://engine.lingo.dev"),
+     *                                "batchSize" (int, default: 25, max: 250),
+     *                                "idealBatchItemSize" (int, default: 250, max: 2500)
+     * @throws InvalidArgumentException If required config is missing or invalid.
+     */
+    public function __construct(array $config)
+    {
+        if (empty($config["apiKey"]) || !is_string($config["apiKey"])) {
+            throw new InvalidArgumentException("Missing or invalid apiKey in configuration.");
+        }
+
+        $this->config = array_merge(self::DEFAULTS, $config);
+
+        // Basic validation for optional parameters
+        if (!filter_var($this->config["apiUrl"], FILTER_VALIDATE_URL)) {
+            throw new InvalidArgumentException("Invalid apiUrl in configuration.");
+        }
+        if (!is_int($this->config["batchSize"]) || $this->config["batchSize"] <= 0 || $this->config["batchSize"] > 250) {
+            throw new InvalidArgumentException("Invalid batchSize in configuration (must be int 1-250).");
+        }
+        if (!is_int($this->config["idealBatchItemSize"]) || $this->config["idealBatchItemSize"] <= 0 || $this->config["idealBatchItemSize"] > 2500) {
```

Comment:
Easier to read here as well:

```suggestion
        if (!is_int($this->config["idealBatchItemSize"]) || $this->config["idealBatchItemSize"] < 1 || $this->config["idealBatchItemSize"] > 2500) {
```